packages/test/test-gc-sweep-tests/src/test/autoRunTest.spec.ts

```diff
@@ -0,0 +1,105 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* eslint-disable max-len */
+import { assert } from "console";
+import {
+    ContainerRuntimeFactoryWithDefaultDataStore,
+} from "@fluidframework/aqueduct";
+import { IRequest } from "@fluidframework/core-interfaces";
+import { ITestObjectProvider } from "@fluidframework/test-utils";
+import { describeNoCompat } from "@fluidframework/test-version-utils";
+import {
+    DefaultSummaryConfiguration,
+    IContainerRuntimeOptions,
+    ISummaryConfigurationHeuristics,
+} from "@fluidframework/container-runtime";
+import { IContainerRuntimeBase } from "@fluidframework/runtime-definitions";
+import { makeRandom } from "@fluid-internal/stochastic-test-utils";
+import { delay } from "@fluidframework/common-utils";
+import { mockConfigProvider } from "../mockConfigProvider";
+import { IgnoreErrorLogger } from "../ignoreErrorLogger";
+import { ContainerManager } from "../containerManager";
+import { rootDataObjectWithChildDataObjectFactory } from "../dataObjectWithChildDataObject";
+import { dataObjectWithCounterFactory } from "../dataObjectWithCounter";
+
+describeNoCompat("GC InactiveObjectX tests", (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+
+    const summaryOptions = DefaultSummaryConfiguration as ISummaryConfigurationHeuristics;
+    summaryOptions.summarizerClientElection = true;
+    // Summaries should run automatically
+    const runtimeOptions: IContainerRuntimeOptions = {
+        summaryOptions,
+        gcOptions: {
+            gcAllowed: true,
+        },
+    };
+    const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+        runtime.IFluidHandleContext.resolveHandle(request);
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        rootDataObjectWithChildDataObjectFactory,
+        [
+            [rootDataObjectWithChildDataObjectFactory.type, Promise.resolve(rootDataObjectWithChildDataObjectFactory)],
+            [dataObjectWithCounterFactory.type, Promise.resolve(dataObjectWithCounterFactory)],
+        ],
+        undefined,
+        [innerRequestHandler],
+        runtimeOptions,
+    );
+
+    // const sessionExpiryDurationMs = 3000; // 3 seconds
+    const inactiveTimeoutMs = 10 * 1000; // 10 seconds
```

Comment:
Something to consider is that you have different timeouts defined in different files that do need to be cohesive (e.g. this and the op delay / ops to wait numbers).  Future idea would be to have a JSON config file that defines all these tunable params in one place and then they get incorporated throughout the test stack.

Reply:
I just wrote a comment down, we'll have to do this in the future